evm/vm/base.py

```diff
@@ -79,14 +78,14 @@ def apply_transaction(self, transaction):
         """
         Apply the transaction to the vm in the current block.
         """
-        computation, block, receipt = self.state.apply_transaction(
+        block, receipt, computation = self.state.apply_transaction(
             transaction,
             self.block,
         )
         self.block = block
-        self.receipts.append(receipt)
+        self.chaindb.persist()
```

Comment:
Yeah, it's gross to have db writes sprinkled about. But... I think we'll have to come back to that one in a future PR.